wire_protocol/src/connection.rs

```diff
@@ -175,6 +175,15 @@ impl Connection for TlsConnection {
             .expect("get connection")
             .write_all(payload)
             .await
+            .context(SendData {
+                endpoint: endpoint.clone(),
+            })?;
+
+        self.stream
+            .as_mut()
+            .expect("get connection")
+            .flush()
```

Comment:
This API Flushes this stream, ensuring that all intermediately buffered contents reach their destination.
https://docs.rs/tokio-rustls/0.22.0/src/tokio_rustls/client.rs.html#93-94 snippet indicates a `TlsStream.flush()` is required to ensure data actually reaches the destination. 
Is this different with a TcpStream ?

Reply:
yeah I think TcpStream itself doesn't have a buffer and any write is directly sent over the wire. However this TlsStream has a buffer inside that needs to be flushed. The name `TlsStream` is kind of misleading.